src/jsMain/js/app/Menu/components/ColorPicker/index.jsx

```diff
@@ -1,52 +1,212 @@
     gadget.color = baaahs.Color.Companion.fromString(hex);
-    this.setState({ gadget });
   };
 
+  _onPickerDragged(ev, data, index) {
+    const { x, y } = data;
+    const { radius, colors } = this.state;
+    const color = xy2rgb(x - radius, y - radius, radius);
+
+    const updatedColors = colors.slice();
+    updatedColors[index] = color;
+
+    this.setState(
+      { selectedIndex: index, grabbingIndex: index, colors: updatedColors },
+      () => this._throttledHandleColorChange()
+    );
+  }
+
+  _onPickerMouseUp(ev, data, index) {
+    const { radius, colors } = this.state;
+    const color = xy2rgb(data.x - radius, data.y - radius, radius);
+
+    const updatedColors = colors.slice();
+    updatedColors[index] = color;
+
+    this.setState(
+      {
+        colors: updatedColors,
+        grabbingIndex: -1,
+      },
+      () => this._handleColorChange()
+    );
+  }
+
+  _renderPicker(picker, index) {
+    const { radius, selectedIndex, grabbingIndex } = this.state;
+    const { color } = picker;
+    const [r, g, b] = color;
+    const position = rgb2xy(color, radius);
+    return (
+      <Draggable
+        key={index}
+        defaultClassName={styles['color-picker--draggable']}
+        defaultClassNameDragging={styles['color-picker--draggable--dragging']}
+        position={{ x: position[0], y: position[1] }}
+        onDrag={(e, data) => {
+          return this._onPickerDragged(e, data, index);
+        }}
+        onStop={(e, data) => {
+          return this._onPickerMouseUp(e, data, index);
+        }}
+        bounds={{ top: 0, left: 0, right: radius * 2, bottom: radius * 2 }}
+      >
+        <div>
+          <div
+            style={{
+              width: pickerRadius * 2,
+              height: pickerRadius * 2,
+              backgroundColor: `rgb(${r}, ${g}, ${b})`,
+            }}
+            className={classNames(styles['color-picker--picker'], {
+              [styles['grabbing']]: index === grabbingIndex,
+              [styles['selected']]: index === selectedIndex,
+            })}
+            onMouseDown={() => {
+              this.setState({
+                selectedIndex: index,
+                grabbingIndex: index,
+              });
+            }}
+          />
+        </div>
+      </Draggable>
+    );
+  }
+
   render() {
-    const { gadget } = this.state;
+    const { radius, colors } = this.state;
 
     return (
-      <div className={styles['color-picker--pad']}>
-        <header>{gadget.name}</header>
-        <SketchPicker
-            color={gadget.color.toHexString()}
-            onChangeComplete={this.handleColorChange}
+      <div
+        ref={(el) => {
+          if (!el) return;
+
+          const { clientWidth } = el;
+          const newRadius = clientWidth / 2;
+          if (radius != newRadius) {
+            this.setState({ radius: newRadius }, () => this.updateWheel());
+          }
+        }}
+        className={styles['color-picker--pad']}
```

Comment:
I thought it would be cool to use old-school css modules. But them seem like too much a hassle over some other css-in-js solution, or outright just use css/sass globally?

What are your thoughts on css modules? Do you think it's something we should stop using, perhaps something like [emotion](https://emotion.sh/docs/introduction)?

Some guy on twitter seems to like emotion: [https://twitter.com/kentcdodds/status/994251975960084480](https://twitter.com/kentcdodds/status/994251975960084480) :)

Reply:
Emotion looks nifty, I haven't heard of it before. I like CSS modules a lot, but something strange is going on and they aren't working as I'd expect them too. I'm used to prefacing things with `:local`, as in `:local(.theNameOfTheClass)`. I like dropping the local, but the nesting with `&` doesn't work as expected, so I might look into that a bit.

Reply:
Oh I removed the sass pre-processor/webpack step due to this kerfuffle: https://github.com/sass/node-sass/issues/2625

But it looks like it got fixed, and we can add it back in? I'll create a GitHub issue to add it back 😄 

I miss sass...